### PR TITLE
/collectors/python.d/varnish: collect smf metrics

### PR DESCRIPTION
**Pull request by ilyam8**
#### Summary

Fixes: #7872

This PR adds [SMF metrics](https://varnish-cache.org/docs/trunk/reference/varnish-counters.html#smf-file-stevedore-counters) (cache on disk) collection to the `varnish` python module.

All available stats:

```
SMF.ssdStorage.c_req                      47686         0.51 Allocator requests
SMF.ssdStorage.c_fail                         0         0.00 Allocator failures
SMF.ssdStorage.c_bytes                668102656      7186.06 Bytes allocated
SMF.ssdStorage.c_freed                140980224      1516.37 Bytes freed
SMF.ssdStorage.g_alloc                    39753          .   Allocations outstanding
SMF.ssdStorage.g_bytes                527122432          .   Bytes outstanding
SMF.ssdStorage.g_space              53159968768          .   Bytes available
SMF.ssdStorage.g_smf                      40130          .   N struct smf
SMF.ssdStorage.g_smf_frag                   311          .   N small free smf
SMF.ssdStorage.g_smf_large                   66          .   N large free smf
```

Collected metrics:

- g_space
- g_bytes

##### Component Name

[/collectors/python.d.plugin/varnish](https://github.com/netdata/netdata/tree/master/collectors/python.d.plugin/varnish)

##### Additional Information

@ibantxo could you test it?

to do it

- download `varnish.chart.py`

> `wget https://raw.githubusercontent.com/ilyam8/netdata/varnish_disk_usage/collectors/python.d.plugin/varnish/varnish.chart.py`

- move it to the `/python.d/` directoty with all other `*.chart.py`

___

Tested by:
- @ibantxo https://github.com/netdata/netdata/pull/7926#issuecomment-580602860
- me https://github.com/netdata/netdata/pull/7926#issuecomment-580638075

### CODE REVIEW

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ collectors/python.d.plugin/varnish/varnish.chart.py  -1
         

Clones added
============
- collectors/python.d.plugin/varnish/varnish.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4917348&bid=16273396)

File: collectors/python.d.plugin/varnish/README.md [hunk 1]
```diff
@@ -1,88 +1,49 @@
...
 
-It produces:
+It uses the `varnishstat` command to provide varnish cache statistics.
 
-1.  **Connections Statistics** in connections/s
+## Requirements
 
-    -   accepted
-    -   dropped
+-   `netdata` user must be a member of `varnish` group
```
Comment (joelhans) [inline collectors/python.d.plugin/varnish/README.md]:
```suggestion
-   `netdata` user must be a member of the `varnish` group 
```

File: collectors/python.d.plugin/varnish/README.md [hunk 2]
```diff
@@ -1,88 +1,49 @@
 # varnish
 
-Module uses the `varnishstat` command to provide varnish cache statistics.
+This module will monitor `Varnish Cache` performance metrics.
```
Comment (joelhans) [inline collectors/python.d.plugin/varnish/README.md]:
```suggestion
This module will monitor [`Varnish Cache`](https://varnish-cache.org/) performance metrics.
```

File: collectors/python.d.plugin/varnish/README.md [hunk 3]
```diff
@@ -1,88 +1,49 @@
...
 
-It produces:
+It uses the `varnishstat` command to provide varnish cache statistics.
```
Comment (joelhans) [inline collectors/python.d.plugin/varnish/README.md]:
```suggestion
It uses the `varnishstat` command to provide Varnish Cache statistics.
```

File: collectors/python.d.plugin/varnish/README.md [hunk 4]
```diff
@@ -1,88 +1,49 @@
...
 
-    -   accepted
-    -   dropped
+-   `netdata` user must be a member of `varnish` group 
 
-2.  **Client Requests** in requests/s
+## Charts
 
-    -   received
+It produces following charts:
```
Comment (joelhans) [inline collectors/python.d.plugin/varnish/README.md]:
```suggestion
This module produces the following charts:
```

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ collectors/python.d.plugin/varnish/varnish.chart.py  -1
         

Clones added
============
- collectors/python.d.plugin/varnish/varnish.chart.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4917348&bid=16273396)

**Comment by ibantxo** [comment]
uouuuu !!! That is working!!! Congratulations!!!

image

**Comment by ilyam8** [comment]
@ibantxo `auto-scaling` units disabled?

**Comment by ibantxo** [comment]
> @ibantxo `auto-scaling` units disabled?

Umm... I think it is enabled:

image

**Comment by ilyam8** [comment]
@ibantxo ok, if you refresh the page, does it still show KiB instead of GiB?

**Comment by ibantxo** [comment]
> @ibantxo ok, if you refresh the page, does it still show KiB instead of GiB?

Yes, have tried with CTRL+F5 and with "invisible mode" @ Chrome.

One thing/doubt: I tried to install via "git clone..." but I had some compilation errors, so... I decided to change the "+ collectors/python.d.plugin/varnish/varnish.chart.py" with the new one and restart netdata. Could it be related to the "auto-scaling" issue?

**Comment by ilyam8** [comment]
yeah, downloading `varnish.chart.py` and replacing old one with new one is better

- `wget https://raw.githubusercontent.com/ilyam8/netdata/varnish_disk_usage/collectors/python.d.plugin/varnish/varnish.chart.py`
- `mv varnish.chart.py /.../python.d/`

However, i copied your output

> https://github.com/netdata/netdata/issues/7872#issuecomment-579697253

that is what i get

Screenshot 2020-01-31 at 11 00 44

**Pull request merged**